### PR TITLE
Parallelize the windows CI to enable us to run all tests all the time

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Split Windows CI into the following Jobs

1. Combine Build and xUnit, which published build artifact
1. Elevated CI tests
1. Unelevated CI test
1. Elevated everything else tests
1. Unelevated everything else tests
1. Package build and test
1. Verify xUnit result

## PR Context  

Before this test:

- A CI run would take ~14 minutes
- A Feature run would take ~30 minutes

After this change, it always runs all tests and takes ~20 minutes.
Job `Unelevated everything else tests` is the long poll, taking ~4 minutes longer than the next shortest job.  We should be able to get this back down to ~15 if we split that test job somehow.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by iSazonov** [comment]
> We should be able to get this back down to ~15 if we split that test job somehow.

By projects?

**Comment by TravisEz13** [comment]
@iSazonov 	I don't want to design the solution right now.  By project, would be a possibility, but optimally, we should try to find a way that makes sense and divides the time close to evenly.

**Comment by iSazonov** [comment]
@TravisEz13 Why we do not select Build in a separate job to use the result for the rest jobs?

Also perhaps we could use [strategy-matrix](https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started-multiplatform?view=azure-devops#add-additional-platforms) to unify process.

**Comment by TravisEz13** [comment]
@iSazonov 	Review the change.  We do the build as a separate job and use the results (artifacts) for the rest of the jobs.

Matrix is less flexible that designing the pipeline yourself.  We would not be able to do the build is a separate job and use the artifacts for the rest of the jobs.

File: tools/ci.psm1 [hunk 1]
```diff
@@ -263,21 +277,48 @@ function Update-TestResults
...
 }
 
+function Invoke-CIxUnit
+{
+    param(
+        [switch]
+        $SkipFailing
+    )
+    $env:CoreOutput = Split-Path -Parent (Get-PSOutput -Options (Get-PSOptions))
+    if(!(Test-Path "$env:CoreOutput\pwsh.exe"))
```
Comment (adityapatwardhan) [inline tools/ci.psm1]: Can we change this to work on Linux and macOS

**Review by adityapatwardhan — APPROVED** [review]

File: tools/ci.psm1 (hunk 1 above)
Comment (iSazonov) [reply tools/ci.psm1]: In the PR xUnit tests run only on Windows. Previously we run the tests on Unix-s too.

**Comment by iSazonov** [comment]
@TravisEz13 Thanks for clarify! I see now how this works.

For reference Setting up parallel jobs, Slicing the test suite
https://docs.microsoft.com/en-us/azure/devops/pipelines/test/parallel-testing-any-test-runner?view=azure-devops#setting-up-parallel-jobs

**Review by PaulHigin — APPROVED** [review]

File: tools/ci.psm1 (hunk 1 above)
Comment (TravisEz13) [reply tools/ci.psm1]: This PR doesn't change Linux or macOS.  I'll deal with this issue when refactoring macOS and Linux

**Pull request merged**